app/models/task/submission_window.rb

```diff
@@ -0,0 +1,37 @@
+require 'bank_holidays'
+
+class Task
+  class SubmissionWindow
+    NUMBER_OF_REPORTING_DAYS = 5
+
+    attr_reader :year, :month
+
+    def initialize(year, month)
+      @month = month
+      @year = year
+    end
+
+    def due_date
+      @due_date ||= last_day_of_submission_window
+    end
+
+    private
+
+    def first_day_of_submission_window
+      Date.new(year, month).end_of_month.next_day
+    end
+
+    def last_day_of_submission_window
+      working_days.take(NUMBER_OF_REPORTING_DAYS).last
+    end
+
+    def working_days
+      dates = Range.new(first_day_of_submission_window, first_day_of_submission_window + 14.days)
```

Comment:
NB: 14 days is a somewhat arbitrary number - including the whole month seemed unnecessary, but you need to include enough days for Easter (double bank holiday) etc.

Reply:
Oops, accidentally resolved this, sorry